src/base/memory_monitor_common.h

```diff
@@ -57,16 +58,15 @@ typedef struct mmon_comp_info
   char name[DB_MAX_IDENTIFIER_LENGTH];
   MMON_OUTPUT_MEM_STAT stat;
   uint32_t num_subcomp;
-  MMON_SUBCOMP_INFO *subcomp_info;
+    std::vector < MMON_SUBCOMP_INFO > subcomp_info;
 } MMON_COMP_INFO;
 
 typedef struct mmon_module_info
 {
-  MMON_SERVER_INFO server_info;
   char name[DB_MAX_IDENTIFIER_LENGTH];
   MMON_OUTPUT_MEM_STAT stat;
   uint32_t num_comp;
-  MMON_COMP_INFO *comp_info;
+    std::vector < MMON_COMP_INFO > comp_info;
```

Comment:
I guess you can remove `num_comp` and `num_subcomp`.

Reply:
Use `/*INDENT-OFF*/` and `/*INDENT-ON*/` for the cpp code in .c file or change its file extension to cpp. We use the different style checkers for cpp and c, and the checker for C doesn't recognize this and may make it ugly. I think it'd be better to use .cpp extension.

Reply:
I think memory monitoring manager needs `num_comp` and `num_comp` because utility needs to know how many components/subcomponents exists for unpacking phase.

Reply:
> I think memory monitoring manager needs num_comp and num_comp because utility needs to know how many components/subcomponents exists for unpacking phase.

I can't find the usage.. maybe will be used in one of next PRs?
I wonder it can't be done with `comp_info.size()`?

When packing or unpacking, the count value is needed. Ok, but I don't understand this structure itself needed the count value. In both side (client and server), the infos are stored in this structure and the structure can know the count with `comp_info.size()`. Isn't it the case? Is there a case where `num_comp` has value, but `comp_info` doesn't have infos?

Reply:
I agree that using the `size()` function of the vector allows us to determine the number of elements inside the vector without needing `num_comp` or `num_subcomp`. However, during the process of unpacking packed information at the client, the client needs to know the count of elements, so it is indeed necessary to include the count information in the sending buffer of server when packing the information. If the struct does not have this count information separated internally, developers may find it inconvenient to pack additional information separately from the information contained in the struct. Since the total amount of data transmitted through the network remains unchanged, I believe it would be more advantageous in terms of readability and manageability to keep this count information.

Reply:
hm.. I don't agree that it's "necessary", but the advantage of removing it is not that serious, and I see you have a point. I'll follow your choice :)